pkg/settings/notifer.go

```diff
@@ -0,0 +1,76 @@
+// Copyright 2023 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package settings
+
+import (
+	"runtime"
+	"runtime/debug"
+
+	"github.com/cockroachdb/cockroach/pkg/util/buildutil"
+	"github.com/cockroachdb/errors"
+)
+
+// NewNotifier is used when a piece of code needs to get a signal when certain
+// settings change. It returns a Notifier associated with the given settings
+// settings; whenever any of these settings change, a non-blocking send on
+// Notifier.Ch() is performed.
+//
+// The Notifier must be Closed.
+func (sv *Values) NewNotifier(settings ...NonMaskedSetting) *Notifier {
+	slots := make([]slotIdx, len(settings))
+	for i := range settings {
+		slots[i] = settings[i].slotIdx()
+	}
+	ch := make(chan struct{}, 1)
```

Comment:
I wonder if we should let the caller make their own ch to choose the buffering capacity (e.g. if they expect to be blocked more or something)? 

Would it make sense to just have `func (v Values) Subscribe(ch chan<- struct{}, settings...) (unsubscribe func())` ?

Reply:
oh, nevermind, I see that's exactly what Andrew proposed in https://github.com/cockroachdb/cockroach/issues/73830#issuecomment-995007403 and you already said you preferred this instead.